water/lib/Dialect/Wave/IR/WaveOps.cpp

```diff
@@ -1456,6 +1604,34 @@ llvm::LogicalResult wave::WriteOp::setIndexFromLattices(
   return llvm::success();
 }
 
+llvm::FailureOr<mlir::ChangeResult> wave::WriteOp::propagateElementsPerThreadForward(
+    llvm::ArrayRef<wave::ElementsPerThreadLatticeValue> operandElements,
+    llvm::MutableArrayRef<wave::ElementsPerThreadLatticeValue> resultElements,
+    llvm::raw_ostream &errs) {
+  // WriteOp has no results, so forward propagation is NoChange
+  return mlir::ChangeResult::NoChange;
+}
+
+llvm::FailureOr<mlir::ChangeResult> wave::WriteOp::propagateElementsPerThreadBackward(
+    llvm::MutableArrayRef<wave::ElementsPerThreadLatticeValue> operandElements,
+    llvm::ArrayRef<wave::ElementsPerThreadLatticeValue> resultElements,
+    llvm::raw_ostream &errs) {
+  // WriteOp: use AttrBasedElementsPerThreadOpTrait logic for register operand
+  // but ignore memory operand (operandElements[1])
+  std::optional<int64_t> elementsPerThread = getElementsPerThread();
+  if (!elementsPerThread)
+    return mlir::ChangeResult::NoChange;
+
+  // Only propagate to operands[0] (register), not operands[1] (memory)
```

Comment:
Trick to avoid hardcoding operand positions: `getRegisterMutable().getOperandNumber()`.